dopamine/discrete_domains/run_experiment.py

```diff
@@ -603,44 +605,67 @@ def _run_iterations(self):
     """
     # TODO(aarg): Change the thread management to an implementation with queues.
     threads = []
+    train_iterations = threading.Semaphore(self._max_simultaneous_iterations)
+    eval_iterations = threading.Semaphore(1)
     # TODO(westurner): See how to refactor the code to avoid setting an internal
     # attribute.
     self._completed_iteration = self._start_iteration
-    for iteration in range(self._start_iteration, self._num_iterations):
-      self._running_iterations.acquire()
-      thread = threading.Thread(
-          target=self._run_one_iteration, args=(iteration,))
+
+    def _start_iteration(*args):
```

Comment:
nit: style guide says to avoid nested functions unless they're nested so that a variable can be closed over.

Reply:
Style guid says: ```They are fine with some caveats. Avoid nested functions or classes except when closing over a local value```

Here we are closing over a local value, aren't we?

This avoids some code repetition.

Reply:
I modified the nested function so that it's not using the `threads` variable.
Does that look better?

Reply:
Sorry -- you're totally right.  `thread` looks like `Thread` looks like `threading` looks like `threads`, and I missed that `thread` <> `threads`.